puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/obal.groovy

```diff
@@ -1,22 +1,19 @@
-def obal(body) {
-
-    def config = [:]
-    body.resolveStrategy = Closure.DELEGATE_FIRST
-    body.delegate = config
-    body()
-
-    def tags = config.tags ? "--tags ${config.tags}" : ""
-    def extra_vars = config.extraVars ?: [:]
+def obal(args) {
     def extra_vars_file = 'extra_vars.yaml'
 
+    tags = args.tags ? "--tags ${args.tags}" : ""
+    extra_vars = args.extraVars ?: [:]
+
     writeYaml file: extra_vars_file, data: extra_vars
     archive extra_vars_file
 
     dir('obal') {
         git url: "https://github.com/theforeman/obal.git", branch: "master"
     }
 
-    wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
-        sh "ANSIBLE_FORCE_COLOR=true PYTHONPATH=obal python -m obal ${tags} -e @${extra_vars_file} ${config.action} ${config.packages}"
+    if (args.packages) {
```

Comment:
This skips past the Obal run if there are no packages to build. We could handle this differently, but for non-package changes this at least gets a green pipeline.

Reply:
I'd argue that "no packages" should result in a "skipped" build, not a "green" one. (But at the same time, green is better than red)